natlas-server/tests/models/test_scope.py

```diff
@@ -0,0 +1,63 @@
+from app import db
+from app.models import ScopeItem
+
+
+def test_new_scope(app):
+    test_scope = ScopeItem(target="10.0.0.0/8", blacklist=False)
+    assert not test_scope.blacklist
+    assert test_scope.target == "10.0.0.0/8"
+    db.session.rollback()
+
+
+def test_add_tags(app):
+    tags = ["test", "tag", "three"]
+    test_scope = ScopeItem(target="127.0.0.1/8", blacklist=False)
+    db.session.add(test_scope)
+    ScopeItem.addTags(test_scope, tags)
+    for tag in test_scope.tags:
+        assert test_scope.is_tagged(tag)
+    db.session.rollback()
+
+
+def test_del_tags(app):
+    tags = ["test", "tags", "three"]
+    test_scope = ScopeItem(target="127.0.0.1/8", blacklist=False)
+    db.session.add(test_scope)
+    ScopeItem.addTags(test_scope, tags)
+    assert len([t.name for t in test_scope.tags]) == 3
+    test_scope.delTag(test_scope.tags[2])
+    assert len([t.name for t in test_scope.tags]) == 2
+    db.session.rollback()
+
+
+def test_get_scope(app):
+    test_scope1 = ScopeItem(target="127.0.0.1/8", blacklist=False)
+    test_scope2 = ScopeItem(target="172.16.0.0/16", blacklist=True)
+    db.session.add(test_scope1)
+    db.session.add(test_scope2)
+    assert len(ScopeItem.getScope()) == 1
+    assert len(ScopeItem.getBlacklist()) == 1
+    db.session.rollback()
+
+
+scopefile = """
+127.0.0.1,one,two,three
+10.11.12.13,four
+10.12.13.14,
+10.13.14.15,one,two
+10.12.13.14
+notevenanip
+"""
+
+
+def test_import_scope(app):
+    f, e, s = ScopeItem.import_scope_list(scopefile.split(), False)
+    assert len(f) == 1
+    assert len(e) == 1
+    assert len(s) == 4
```

Comment:
This is not very intuitive. What are these return values?

Reply:
fail, exists, succeeds. I'm not particularly fond of this "return three lists" approach but didn't want to change logic while submitting tests.